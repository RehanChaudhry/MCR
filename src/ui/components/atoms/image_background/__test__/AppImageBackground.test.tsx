import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import {
  AppImageBackground,
  CONTAINER_TYPES
} from "ui/components/atoms/image_background/AppImageBackground";

test("snapshot testing", () => {
  const rendered = render(
    <AppImageBackground
      containerShape={CONTAINER_TYPES.CIRCLE}
      icon={require("assets/images/check.png")}
    />
  ).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("renders icon correctly", () => {
  const { queryByTestId } = render(
    <AppImageBackground
      containerShape={CONTAINER_TYPES.CIRCLE}
      icon={require("assets/images/check.png")}
    />
  );
  let icon = queryByTestId("icon");
  expect(icon).not.toBeNull();
});

it("should properly perform click event", () => {
  const onPress = jest.fn();
  const { getByTestId } = render(
    <AppImageBackground
      containerShape={CONTAINER_TYPES.CIRCLE}
      icon={require("assets/images/check.png")}
      onPress={onPress}
    />
  );
  const imageButton = getByTestId("image-container");
  fireEvent.press(imageButton);
  expect(onPress).toBeCalledTimes(1);
});
