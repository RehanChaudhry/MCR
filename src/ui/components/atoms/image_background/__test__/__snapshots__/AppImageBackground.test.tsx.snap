// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 60.466019417475735,
        "justifyContent": "center",
        "width": 60.466019417475735,
      },
      Object {
        "backgroundColor": "#E5E7EB",
        "shadowColor": "#E5E7EB",
      },
      Object {
        "borderRadius": 50,
      },
      undefined,
    ]
  }
>
  20
</View>
`;
